### DISCUSSION TITLE
Request to delete and recreate empty columns when changing between incompatible data types

### DISCUSSION THREAD

**Discussion by szmarczak** [supabase/supabase#33096] [Feature Requests]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Title.

## To Reproduce

1. Create a column with int8 type:

![image](https://github.com/user-attachments/assets/022b5fa8-7a87-4096-88bc-c05a3c0f9566)

2. Try to change it to uuid:

![image](https://github.com/user-attachments/assets/cd2792d3-b4d8-4d2b-b31c-5bcd12302631)

## Expected behavior

No error.

## Screenshots

Attahed,

## System information

- OS: Windows
- Browser (if applies) Edge
- Version of supabase-js: N/A
- Version of Node.js: N/A

**Comment by GaryAustin1** [comment]
In general you can't directly change a uuid type to an int type or opposite, especially if data exists.  But even without it is an issue.
Easiest just to delete and create a new column.  
https://stackoverflow.com/questions/61027222/postgresql-change-column-type-from-uuid-to-int4
https://stackoverflow.com/questions/20342717/postgresql-change-column-type-from-int-to-uuid

**Comment by szmarczak** [comment]
Well, couldn't it just do that (remove & create) automatically if there's no data? This is more confusing than the limited options in the column type select box.

**Comment by GaryAustin1** [comment]
I imagine if they (Supabase) were to prioritize that it could be done.  It certainly gets more complicated if data exists in the rows as everything would need to be converted.  But it is currently not a feature and Postgres itself can't do it directly either with just an alter column.

**Comment by encima** [comment]
Thanks for reporting, have updated this to be a feature request for the frontend team 🙏

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
